### PR TITLE
Couchforms dbaccessors cleanup

### PR DESCRIPTION
**Pull request by dannyroberts**

### CODE REVIEW

**Comment by dannyroberts** [comment]
@proteusvacuum @orangejenny

**Comment by proteusvacuum** [comment]
This is maybe a silly question (and might be the wrong place to ask), but won't this potentially fail hard in production (and not get caught by unit tests?). Like, if I use one of those methods in regular code, I write really good unit tests which all pass, then push to prod? Or is that just a danger we're OK to live with? 

Maybe adding a log message to the `unit_testing_only` decorator that will print this into my test window might alleviate that danger somewhat. What do you think?

(Also, merging, 'cause PR looks good).

File: corehq/ex-submodules/couchforms/dbaccessors.py [hunk 1]
```diff
@@ -102,5 +98,5 @@ def get_exports_by_form(domain):
         startkey=[domain],
         endkey=[domain, {}],
         group=True,
-        stale=settings.COUCH_STALE_QUERY
+        stale=stale_ok()
```
Comment (czue) [inline corehq/ex-submodules/couchforms/dbaccessors.py]: should this be moved to `analytics` if it's going to return stale results?

**Comment by dannyroberts** [comment]
@proteusvacuum, yeah that's a conundrum. You really want to assert that it's being called directly from a unit test file, or directly from a function only ever called by a unit test file, but I imagine that's a relatively tricky thing to get right. We've never so far had a problem with people by mistake using a unit_testing_only function in production. My stance is that either it just won't ever happen, or when it does we'll learn something about the circumstance under which it became possible for it to be overlooked and put better safety mechanisms in place. (I accept the criticism here that if there are obvious safety measures to put in place before a problem happens, that's even better.) I think the suggestion you made actually wouldn't help; since tons of test call these functions, we'd get really used to seeing these print out, so it wouldn't stand out the one time it matters.

For what it's worth, if you ever run the code through runserver and hitting a page, it'll certainly fail obviously; granted a lot of our backend code relies much more on automated testing than on manual testing, which is a good thing. So you're right that there's a potential problem here. I currently think it's small enough not to address, but I'm open to other view points.

**Comment by proteusvacuum** [comment]
Thanks for the explanation. Sounds pretty reasonable to hold off on putting
more safety mechanisms in place, especially if we've never run into an
issue with it in the past.

Another option in the case of these dbacessors would be to just add them to
the test module `hqadmin.test.dbaccessors` for example. That way if you
wanted to use them, you'd have to at least import it from the test
directory, which would be an extra red flag if you were to use them in
production code (it also nicely separates test code from prod code, which I
think is an added benefit).

**Comment by dannyroberts** [comment]
well that certainly sounds reasonable; I'd be in favor of moving in that direction

**Comment by czue** [comment]
> Another option in the case of these dbacessors would be to just add them to the test module `hqadmin.test.dbaccessors` for example

image

**Pull request merged**